Fastly-Tls-PrivateKeys/fastly-tls-privatekeys.json

```diff
@@ -0,0 +1,140 @@
+                    "$ref": "#/definitions/KeyType"
+                },
+                "KeyLength": {
+                    "$ref": "#/definitions/KeyLength"
+                },
+                "PublicKeySha1": {
+                    "$ref": "#/definitions/PublicKeySha1"
+                },
+                "Replace": {
+                    "$ref": "#/definitions/Replace"
+                },
+                "CreatedAt": {
+                    "$ref": "#/definitions/CreatedAt"
+                }
+            },
+            "required": [
+                "Key"
+            ]
+        },
+        "CreatedAt": {
+            "description": "Date and time in ISO 8601 format. Read-only.",
+            "type": "string",
+            "format": "date-time"
+        },
+        "DeletedAt": {
+            "description": "Date and time in ISO 8601 format. Read-only.",
+            "type": "string",
+            "format": "date-time"
+        },
+        "KeyType": {
+            "type": "string",
+            "description": "The algorithm used to generate the private key. Must be RSA"
+        },
+        "KeyLength": {
+            "type": "number",
+            "description": "The key length used to generate the private key"
+        },
+        "PublicKeySha1": {
+            "type": "string",
+            "description": "Useful for safely identifying the key"
+        },
+        "Replace": {
+            "type": "boolean",
+            "description": "A recommendation from Fastly to replace this private key and all associated certificates"
+        }
+    },
+    "typeConfiguration": {
+        "properties": {
+          "FastlyAccess": {
+            "$ref": "#/definitions/FastlyAccess"
+          }
+        },
+        "additionalProperties": false,
+        "required": [
+          "FastlyAccess"
+        ]
+    },
+    "properties": {
+        "Type": {
+            "type": "string",
+            "description": "Resource type. [Default tls_private_key]",
+            "enum": [
+                "tls_private_key"
+            ]
+        },
+        "Id": {
+            "type": "string",
+            "description": "Alphanumeric string identifying a private Key"
+        },
+        "Attributes": {
+            "$ref": "#/definitions/Attributes"
+        }
+    },
+    "additionalProperties": false,
+    "required": [
+        "Attributes"
+    ],
+    "readOnlyProperties": [
+        "/properties/Id",
+        "/properties/Attributes/KeyType",
+        "/properties/Attributes/KeyLength",
+        "/properties/Attributes/PublicKeySha1",
+        "/properties/Attributes/Replace"
+    ],
+    "writeOnlyProperties": [
+        "/properties/Attributes/Key"
+    ],
+    "primaryIdentifier": [
+        "/properties/Id"
+    ],
+    "handlers": {
+        "create": {
+            "permissions": []
+        },
+        "read": {
+            "permissions": []
+        },
+        "update": {
+            "permissions": []
+        },
```

Comment:
This should be removed as the resource update handle throws a `exceptions.NotUpdatable();`